sensor_msgs/include/sensor_msgs/point_cloud_conversion.h

```diff
@@ -115,15 +116,6 @@ static inline bool convertPointCloudToPointCloud2 (const sensor_msgs::PointCloud
     */ 
 static inline bool convertPointCloud2ToPointCloud (const sensor_msgs::PointCloud2 &input, sensor_msgs::PointCloud &output)
 {
-  // Get all fields size and check if we have non-float32 values
-  for (size_t d = 0; d < input.fields.size (); ++d)
-  {
-    if (input.fields[d].datatype != sensor_msgs::PointField::FLOAT32)
-    {
-      std::cerr << boost::str(boost::format("sensor_msgs::PointCloud accepts only float32 values, but field %d (%s) has field type %d!")%(int)d% input.fields[d].name%input.fields[d].datatype) << std::endl;
```

Comment:
Removing that condition would change the behavior: before, RGB as uchar could not be put into a PointCloud (by being converted to float32). Is that a behavior we want ? @tfoote , probably not